agent/functional_tests/tests/functionaltests_test.go

```diff
@@ -653,6 +654,79 @@ func telemetryTestWithStatsPolling(t *testing.T, taskDefinition string) {
 	require.NoError(t, err, "Waiting for task stop failed")
 }
 
+func telemetryStorageStatsTest(t *testing.T, taskDefinition string) {
+	newClusterName := "ecstest-storagestats-" + uuid.New()
+	putAccountInsights := ecsapi.PutAccountSettingInput{
+		Name:  aws.String("containerInsights"),
+		Value: aws.String("enabled"),
+	}
+	_, err := ECS.PutAccountSetting(&putAccountInsights)
+	require.NoError(t, err, "Failed to update account settings")
+
+	_, err = ECS.CreateCluster(&ecsapi.CreateClusterInput{
+		ClusterName: aws.String(newClusterName),
+	})
+	require.NoError(t, err, "Failed to create cluster")
+	defer DeleteCluster(t, newClusterName)
+
+	agentOptions := AgentOptions{
+		ExtraEnvironment: map[string]string{
+			"ECS_CLUSTER": newClusterName,
+		},
+	}
+	agent := RunAgent(t, &agentOptions)
+	defer agent.Cleanup()
+
+	cwclient := cloudwatch.New(session.New(), aws.NewConfig().WithRegion(*ECS.Config.Region))
+	params := &cloudwatch.GetMetricStatisticsInput{
+		Namespace: aws.String("ECS/ContainerInsights"),
+		Period:    aws.Int64(60),
+		Statistics: []*string{
+			aws.String("Average"),
+			aws.String("SampleCount"),
+		},
+		Dimensions: []*cloudwatch.Dimension{
+			{
+				Name:  aws.String("ClusterName"),
+				Value: aws.String(newClusterName),
+			},
+		},
+	}
+
+	// start storageStats task
+	testTask, err := agent.StartTask(t, taskDefinition)
+	require.NoError(t, err, "Failed to start storageStats task")
+	err = testTask.WaitRunning(waitTaskStateChangeDuration)
+	require.NoError(t, err, "Error wait storageStats task running")
+
+	// collect and validate minimal state metrics
+	time.Sleep(waitMinimalMetricsInCloudwatchDuration)
```

Comment:
Does it mean we are validating one data metric (5 minutes)? Should we increase it to 6 minutes(something more than 5?), if there is a slight delay to post metrics to CW?

Reply:
I ran this test 20 times and it succeeded 20 times.  
We can bump this time up if we encounter flakiness, but I see at least 2 datapoints, sometimes three in my responses.  We are only validating that datapoints exist with this test.